// tdesign-vue-next/es/color-picker/color-picker-panel.mjs
/**
 * tdesign v1.9.5
 * (c) 2024 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-2af2f3cc.mjs';
import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import props from './props.mjs';
import ColorPanel from './panel/index.mjs';
import { usePrefixClass } from '../hooks/useConfig.mjs';
import '../_chunks/dep-0659d768.mjs';
import { _ as _arrayMap } from '../_chunks/dep-5b49f687.mjs';
import { _ as _baseIteratee } from '../_chunks/dep-b29cd8fe.mjs';
import { _ as _basePickBy } from '../_chunks/dep-30c143bf.mjs';
import { _ as _getAllKeysIn } from '../_chunks/dep-4fb87459.mjs';
import '../_chunks/dep-ca297029.mjs';
import '../_chunks/dep-0714c768.mjs';
import '../_chunks/dep-26ed0cac.mjs';
import '../_chunks/dep-5196116b.mjs';
import './const.mjs';
import '../_common/js/color-picker/constants.mjs';
import './panel/header.mjs';
import '../radio/index.mjs';
import '../radio/radio.mjs';
import '../utils/helper.mjs';
import '../_chunks/dep-bc36f9a0.mjs';
import '../_chunks/dep-8fde58dd.mjs';
import '../_chunks/dep-4ab8ef1f.mjs';
import '../_chunks/dep-dee62ba9.mjs';
import '../_chunks/dep-91b73ccb.mjs';
import '../_chunks/dep-1bbef55d.mjs';
import '../_chunks/dep-eb726c0b.mjs';
import '../_chunks/dep-9672d334.mjs';
import '../_chunks/dep-44a188ed.mjs';
import '../_chunks/dep-f731006d.mjs';
import '../_chunks/dep-de53e2e8.mjs';
import '../_chunks/dep-78b9feac.mjs';
import '../_chunks/dep-6bcf8998.mjs';
import '../radio/props.mjs';
import '../radio/constants.mjs';
import '../hooks/useDisabled.mjs';
import '../_chunks/dep-3b219639.mjs';
import '../hooks/useVModel.mjs';
import '../_chunks/dep-e6085593.mjs';
import '../hooks/tnode.mjs';
import '../_chunks/dep-c35bb50c.mjs';
import '../_chunks/dep-79da3af4.mjs';
import '../_chunks/dep-e242a06a.mjs';
import '../utils/render-tnode.mjs';
import '../_chunks/dep-91bceb4f.mjs';
import '../_chunks/dep-8ad6255c.mjs';
import '../_chunks/dep-5f0aedd7.mjs';
import '../_chunks/dep-5cb3d4ea.mjs';
import '../_chunks/dep-ea111861.mjs';
import '../_chunks/dep-2d33c97a.mjs';
import '../_chunks/dep-fe843970.mjs';
import '../_chunks/dep-ac0f9138.mjs';
import '../_chunks/dep-b76458c6.mjs';
import '../config-provider/useConfig.mjs';
import '../_chunks/dep-8f5d66bf.mjs';
import '../_chunks/dep-766c643b.mjs';
import '../_chunks/dep-2a05de08.mjs';
import '../_chunks/dep-4ee5e119.mjs';
import '../_chunks/dep-b59c8c62.mjs';
import '../_chunks/dep-183c499b.mjs';
import '../_chunks/dep-4e2e86d1.mjs';
import '../_chunks/dep-616c78b0.mjs';
import '../_chunks/dep-8ce99060.mjs';
import '../_chunks/dep-39423da8.mjs';
import '../_chunks/dep-de4bb4cb.mjs';
import '../_chunks/dep-f9e3f698.mjs';
import '../_chunks/dep-5654af14.mjs';
import '../_common/js/global-config/default-config.mjs';
import '../_common/js/global-config/locale/zh_CN.mjs';
import '../_chunks/dep-2b12f580.mjs';
import '../_chunks/dep-2a58ca94.mjs';
import '../_chunks/dep-115707a4.mjs';
import '../config-provider/type.mjs';
import '../_chunks/dep-c1b8140c.mjs';
import '../_chunks/dep-7d3c27b1.mjs';
import '../radio/group.mjs';
import '../_chunks/dep-beb354f3.mjs';
import '../_chunks/dep-c8d3a4cf.mjs';
import '../_chunks/dep-670ff068.mjs';
import '../_chunks/dep-41d56db9.mjs';
import '../_chunks/dep-3d661c92.mjs';
import '../radio/radio-group-props.mjs';
import '../radio/useKeyboard.mjs';
import '../utils/dom.mjs';
import '../utils/easing.mjs';
import '../_common/js/common.mjs';
import '../watermark/hooks.mjs';
import '../hooks/useResizeObserver.mjs';
import '../radio/radio-button.mjs';
import '../utils/withInstall.mjs';
import './style/css.mjs';
import '../radio/type.mjs';
import './hooks.mjs';
import './panel/linear-gradient.mjs';
import './utils/index.mjs';
import '../_common/js/color-picker/index.mjs';
import '../_common/js/color-picker/cmyk.mjs';
import '../_common/js/color-picker/color.mjs';
import '../_chunks/dep-db66fb97.mjs';
import '../_chunks/dep-83ed1aa0.mjs';
import '../_common/js/color-picker/gradient.mjs';
import '../_common/js/color-picker/draggable.mjs';
import '../input-number/index.mjs';
import '../input-number/input-number.mjs';
import 'tdesign-icons-vue-next';
import '../button/index.mjs';
import '../button/button.mjs';
import '../loading/index.mjs';
import '../loading/loading.mjs';
import '../loading/icon/gradient.mjs';
import '../_common/js/loading/circle-adapter.mjs';
import '../_common/js/utils/set-style.mjs';
import '../_common/js/utils/helper.mjs';
import '../loading/props.mjs';
import '../hooks/useTeleport.mjs';
import '../loading/type.mjs';
import '../loading/plugin.mjs';
import '../_chunks/dep-546d3f9b.mjs';
import '../_chunks/dep-e33169b8.mjs';
import '../_chunks/dep-0b5ca4b8.mjs';
import '../_chunks/dep-fe99cdb6.mjs';
import '../_chunks/dep-f5778e4b.mjs';
import '../_chunks/dep-7fab1039.mjs';
import '../_chunks/dep-865d41b7.mjs';
import '../button/props.mjs';
import '../hooks/useRipple.mjs';
import '../hooks/useKeepAnimation.mjs';
import '../utils/set-style.mjs';
import '../button/type.mjs';
import '../input/index.mjs';
import '../input/input.mjs';
import '../input/props.mjs';
import '../hooks/useGlobalIcon.mjs';
import '../input/useInput.mjs';
import '../form/const.mjs';
import '../hooks/index.mjs';
import '../hooks/icon.mjs';
import '../hooks/slot.mjs';
import '../hooks/useCommonClassName.mjs';
import '../hooks/useDefaultValue.mjs';
import '../hooks/useDestroyOnClose.mjs';
import '../hooks/useVirtualScroll.mjs';
import '../hooks/useImagePreviewUrl.mjs';
import '../_common/js/upload/utils.mjs';
import '../_common/js/log/log.mjs';
import '../input/useLengthLimit.mjs';
import '../_common/js/log/index.mjs';
import '../input/useInputEventHandler.mjs';
import '../input/useInputWidth.mjs';
import '../input/input-group.mjs';
import '../input/input-group-props.mjs';
import '../input/type.mjs';
import '../input-number/props.mjs';
import '../input-number/useInputNumber.mjs';
import '../_common/js/input-number/number.mjs';
import '../_common/js/input-number/large-number.mjs';
import '../input-number/type.mjs';
import './panel/base-props.mjs';
import './panel/saturation.mjs';
import './panel/hue.mjs';
import './panel/slider.mjs';
import './panel/alpha.mjs';
import './panel/format/index.mjs';
import '../select/index.mjs';
import '../select/select.mjs';
import '../_chunks/dep-a13534db.mjs';
import '../_chunks/dep-4c5036fd.mjs';
import '../common-components/fake-arrow.mjs';
import '../select-input/index.mjs';
import '../select-input/select-input.mjs';
import '../popup/index.mjs';
import '../popup/popup.mjs';
import '@popperjs/core';
import '../popup/container.mjs';
import '../popup/props.mjs';
import '../popup/type.mjs';
import '../select-input/props.mjs';
import '../select-input/useSingle.mjs';
import '../select-input/useMultiple.mjs';
import '../tag-input/index.mjs';
import '../tag-input/tag-input.mjs';
import '../tag-input/props.mjs';
import '../tag-input/hooks/useTagScroll.mjs';
import '../tag-input/useTagList.mjs';
import '../tag/index.mjs';
import '../tag/tag.mjs';
import '../tag/props.mjs';
import '../tag/check-tag.mjs';
import '../tag/check-tag-props.mjs';
import '../tag/check-tag-group.mjs';
import '../tag/check-tag-group-props.mjs';
import '../tag/type.mjs';
import '../tag-input/hooks/useHover.mjs';
import '../tag-input/hooks/useDragSorter.mjs';
import '../tag-input/type.mjs';
import '../select-input/useOverlayInnerStyle.mjs';
import '../_chunks/dep-fc25c249.mjs';
import '../_chunks/dep-0dc95788.mjs';
import '../select-input/type.mjs';
import '../select/select-panel.mjs';
import '../_chunks/dep-7966403d.mjs';
import '../_chunks/dep-8ea10f6c.mjs';
import '../select/option.mjs';
import '../select/option-props.mjs';
import '../checkbox/index.mjs';
import '../checkbox/checkbox.mjs';
import '../checkbox/props.mjs';
import '../checkbox/constants.mjs';
import '../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../_common/js/utils/observe.mjs';
import '../checkbox/hooks/useKeyboardEvent.mjs';
import '../checkbox/group.mjs';
import '../checkbox/checkbox-group-props.mjs';
import '../checkbox/type.mjs';
import '../select/helper.mjs';
import '../select/optionGroup.mjs';
import '../select/option-group-props.mjs';
import '../select/props.mjs';
import '../select/hooks/usePanelVirtualScroll.mjs';
import '../select/hooks/useSelectOptions.mjs';
import '../select/hooks/useKeyboardControl.mjs';
import '../select/type.mjs';
import './panel/format/inputs.mjs';
import './panel/format/config.mjs';
import './panel/swatches.mjs';

var arrayMap = _arrayMap,
  baseIteratee = _baseIteratee,
  basePickBy = _basePickBy,
  getAllKeysIn = _getAllKeysIn;

/**
 * Creates an object composed of the `object` properties `predicate` returns
 * truthy for. The predicate is invoked with two arguments: (value, key).
 *
 * @static
 * @memberOf _
 * @since 4.0.0
 * @category Object
 * @param {Object} object The source object.
 * @param {Function} [predicate=_.identity] The function invoked per property.
 * @returns {Object} Returns the new object.
 * @example
 *
 * var object = { 'a': 1, 'b': '2', 'c': 3 };
 *
 * _.pickBy(object, _.isNumber);
 * // => { 'a': 1, 'c': 3 }
 */
function pickBy(object, predicate) {
  if (object == null) {
    return {};
  }
  var props = arrayMap(getAllKeysIn(object), function (prop) {
    return [prop];
  });
  predicate = baseIteratee(predicate);
  return basePickBy(object, props, function (value, path) {
    return predicate(value, path[0]);
  });
}
var pickBy_1 = pickBy;

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var _ColorPickerPanel = defineComponent({
  name: "TColorPickerPanel",
  inheritAttrs: false,
  props: _objectSpread({}, props),
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var newProps = computed(function () {
      return pickBy_1(_objectSpread(_objectSpread({}, props2), attrs), function (v) {
        return v !== void 0;
      });
    });
    var prefix = usePrefixClass();
    return function () {
      return createVNode(ColorPanel, mergeProps(newProps.value, {
        "popupProps": null,
        "close-btn": false,
        "class": "".concat(prefix.value, "-is-inline")
      }), null);
    };
  }
});

export { _ColorPickerPanel as default };
